oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/metrics/Metrics.java

```diff
@@ -80,12 +81,21 @@
     public abstract Metrics toDay();
 
     /**
-     * Extend the {@link #survivalTime}
+     * Set the last update timestamp
      *
-     * @param value to extend
+     * @param timestamp to extend
      */
-    public void extendSurvivalTime(long value) {
-        survivalTime += value;
+    public void setLastUpdateTimestamp(long timestamp) {
+        lastUpdateTimestamp += timestamp;
+    }
+
+    /**
+     * @param timestamp        of current time
+     * @param expiredThreshold represents the duration between last update time and the time point removing from cache.
+     * @return true means this metrics should be removed from cache.
+     */
+    public boolean isSurvival(long timestamp, long expiredThreshold) {
```

Comment:
I think `Survival` is more suitable to say the metrics should be **cached** instead of **removed**, what about renaming `isSurvival` to `isExpired`?

Reply:
Make sense.